### PR TITLE
runCommandNoCC -> runCommand deprecation

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Simplify to remove doubt when writing and reviewing expressions.

This produces no rebuilds except in the yarn2nix-moretea package, which is updated to use `runCommand`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/trivial-builders.nix [hunk 1]
```diff
@@ -24,16 +24,16 @@ rec {
   * `allowSubstitutes = false;`
   * to a derivation’s attributes.
   */
-  runCommand = runCommandNoCC;
-  runCommandLocal = runCommandNoCCLocal;
+  runCommandNoCC = lib.warn "runCommandNoCC is deprecated. Use the equivalent runCommand function instead." runCommand;
```
Comment (andir) [inline pkgs/build-support/trivial-builders.nix:27]: Do we plan removal at some point in the future? Now would be the time to announce that we will drop in in ~2022.11?

Comment (roberth) [reply pkgs/build-support/trivial-builders.nix]: I'd say it announces itself. No need to use up more brain bandwidth than necessary.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 134225` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>yarn2nix</li>
  </ul>
</details>

File: pkgs/build-support/trivial-builders.nix (hunk 1 above)
Comment (siraben) [reply pkgs/build-support/trivial-builders.nix]: Though, maybe link to the PR like we did for deprecating `stdenv.lib`. Not necessary though.

**Comment by roberth** [comment]
Oh no, I was still looking into making ofborg accept this.

**Comment by roberth** [comment]
I'll make a PR to fix ofborg eval.

**Comment by roberth** [comment]
The deprecation warning has been reverted in #134230, making this PR ineffective.

We can't have deprecate it until https://github.com/NixOS/ofborg/issues/575, unless someone knows a workaround to please / trick ofborg.

**Pull request merged**